explainer.md

```diff
@@ -73,95 +75,28 @@ A LaunchEvent containing a sequence<[FileSystemFileHandle](https://github.com/WI
-interface FileLaunchEvent {
-  cause: 'file';
-  files: FileSystemBaseHandle[];
-}
-
-// Caused when the browser would have navigated to an in-app link.
-interface UrlLaunchEvent {
-  cause: 'url';
-  request: FetchAPIRequest;
-}
-
-type LaunchEvent = IconLaunchEvent | FileLaunchEvent | UrlLaunchEvent;
-
-// Example Handler
-const launchEventHandler = (event: LaunchEvent) => {
-  if (event.cause === 'file') {
-    // Do something with files.
-  }
-};
-
-```
-
-More detail is available on the [sw-launch](https://github.com/WICG/sw-launch/blob/master/explainer.md) repository.
-
 ### Concerns
 
-This proposal (with new service worker events) was presented to the Service Worker and Web Platform working groups at TPAC 2018. There was concern that executing more code in the service worker might affect performance, and it may not be clear to users which web application is running code, if no client window has been given focus yet.
-
-A different API, was considered with flow similar to the following:
-1. If no window exists, create a new one
-2. Fire launch event on first active window
-
-However, this API would require jumping through some hoops in common cases. Consider a document editor:
-1. A user is editing document1.doc
-2. User then opens document2.doc
-   1. A launch event on document1.doc is fired
-   2. As the editor doesn't want to lost the changes to document1, a message is posted to the service worker.
-      1. The service worker checks to see if document2.doc is already open
-         1. If yes, focus that window
-         2. If no, create a new client, post it the document and focus the client.
-
-Compare to having the event on the service worker, where we can cut out the first postMessage
-1. A user is editing document1.doc
-2. User then opens document2.doc
-   1. A launch event is fired on the service worker
-      1. The service worker checks to see if document2 is already open
-         1. If yes, focus that window
-         2. If no, create a new client, post it the document and focus the client.
-
-Which API is better depends largely on what the more common case is likely to be:
-1. Opening the file in an existing Window
-   - Note: This only seems to make sense in apps that can only have one instance, otherwise the existing window the browser picks is likely to be fairly arbitrary.
-2. Opening the file in a new window
-
-Arguably, the second case is more common on current desktop operating systems. 
-
-#### Will the current applications behavior be supported?
-Below is a not-at-all scientific collection of how a few common apps handle files being opened.
+#### Will the current application behavior be supported?
+Below is a not-at-all scientific collection of how a few common apps handle files being opened. SW Launch refers to the case where we fire a launch event on the service worker, while client launch refers to a theoretical event handler on the client window.
 
 App     | SW Launch   | Client Launch   | Description
```

Comment:
This is a really good analysis but I think it should also probably move into sw-launch (and be removed from here). Even though it's related to how files open, it would impact the design of launch event as opposed to file handlers. WDYT?